src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingChannelMapping.java

```diff
@@ -0,0 +1,123 @@
+import static uk.gov.hmcts.reform.sscs.ccd.domain.YesNo.isYes;
+import static uk.gov.hmcts.reform.sscs.reference.data.model.HearingChannel.FACE_TO_FACE;
+import static uk.gov.hmcts.reform.sscs.reference.data.model.HearingChannel.NOT_ATTENDING;
+import static uk.gov.hmcts.reform.sscs.reference.data.model.HearingChannel.PAPER;
+import static uk.gov.hmcts.reform.sscs.reference.data.model.HearingChannel.TELEPHONE;
+import static uk.gov.hmcts.reform.sscs.reference.data.model.HearingChannel.VIDEO;
+
+@Slf4j
+@SuppressWarnings("PMD.CyclomaticComplexity")
+public final class HearingChannelMapping {
+
+    private HearingChannelMapping() {
+
+    }
+
+    public static List<HearingChannel> getHearingChannels(@Valid SscsCaseData caseData) {
+        return List.of(getHearingChannel(caseData));
+    }
+
+    public static HearingChannel getHearingChannel(@Valid SscsCaseData caseData) {
+        if (isYes(caseData.getDwpIsOfficerAttending())) {
+            return FACE_TO_FACE;
+        }
+
+        List<HearingChannel> hearingChannels = getAllHearingChannelPreferences(caseData);
+
+        if (hearingChannels.contains(FACE_TO_FACE)) {
+            return FACE_TO_FACE;
+        } else if (hearingChannels.contains(VIDEO)) {
+            return VIDEO;
+        } else if (hearingChannels.contains(TELEPHONE)) {
+            return TELEPHONE;
+        } else {
+            return PAPER;
+        }
+    }
+
+
+    public static List<HearingChannel> getAllHearingChannelPreferences(@Valid SscsCaseData caseData) {
+
+        HearingChannel individualPreferredHearingChannel = getIndividualPreferredHearingChannel(
+            caseData.getAppeal().getHearingSubtype(),
+            caseData.getAppeal().getHearingOptions()
+        );
+
+        List<HearingChannel> hearingChannels = new ArrayList<>();
+
+        hearingChannels.add(individualPreferredHearingChannel);
+
+        if (nonNull(caseData.getOtherParties())) {
+            hearingChannels.addAll(caseData.getOtherParties().stream()
+                .map(CcdValue::getValue)
+                .map(otherParty -> getIndividualPreferredHearingChannel(
+                    otherParty.getHearingSubtype(),
+                    otherParty.getHearingOptions()))
+                .collect(Collectors.toList()));
+        }
+
+        return hearingChannels;
+    }
+
+    public static List<String> getHearingChannelsHmcReference(@Valid SscsCaseData caseData) {
+        return getHearingChannels(caseData).stream()
+            .map(HearingChannel::getHmcReference)
+            .collect(Collectors.toList());
+    }
+
+    public static HearingChannel getIndividualPreferredHearingChannel(HearingSubtype hearingSubtype, HearingOptions hearingOptions) {
+        if (isNull(hearingOptions)) {
+            return null;
+        }
+
+        if (isFalse(hearingOptions.isWantsToAttendHearing())) {
+            return NOT_ATTENDING;
+        }
+
+        if (nonNull(hearingSubtype)) {
```

Comment:
Can be simplified. By using isNull(caseData.getOtherParties()) and return null in that case you can reduce the nesting and therefore cyclomatic complexity.

Reply:
It would seem this line was added prior to Ivan taking over and finishing up the changes.
Given the circumstances I'd rather get this through so I'll approve the current status and approve the review.